src/bin/nydusd/main.rs

```diff
@@ -363,6 +373,83 @@ fn handle_rlimit_nofile_option(args: &ArgMatches, option_name: &str) -> Result<(
     Ok(())
 }
 
+fn send_unix_socket_request(socket_path: &str, method: &str, url: &str) -> Option<String> {
```

Comment:
How about putting this into `api/src/config.rs`, and naming it `fn from_source(...)`.